cupy/cuda/memory.pyx

```diff
@@ -541,6 +541,13 @@ cpdef MemoryPointer malloc_managed(size_t size):
 cdef object _current_allocator = _malloc
 cdef object _thread_local = threading.local()
 
+
+def _set_thread_local_allocator(allocator):
+    previous_allocator = getattr(_thread_local, 'allocator', None)
```

Comment:
Would it make sense to split this part out into a `_get_thread_local_allocator`?

Reply:
Thanks! I agree that's better. Fixed in 4d27577a0ea1b680ae30c2b688906348e6c57c11
Also rewrote using `except AttributeError` as it's faster than `getattr`.